Sources/Parser.swift

```diff
@@ -71,8 +71,24 @@ public class TokenParser {
     tokens.insert(token, at: 0)
   }
 
+  public func compileFilter(_ token: String) throws -> Resolvable {
```

Comment:
Why do these methods (`compileFilter`, `compileExpression` and `compileResolvable`) have to be public?

Note: if they can be internal, don't forget to reset the `Expression` protocol back to internal.

Reply:
Ehm, do they need to be used by 3rd party extensions? Do you have a scenario where that'd be necessary?

Reply:
I cant think of any now but if I needed it in `filter` then it might be needed in something else complex too. Anyway, I don't see any harm in making these methods public.

Reply:
Honestly, a library should only expose the minimal surface necessary for a developer to use it. If at any point, something is missing, it can be added later on.

Maybe someone else has some thoughts on this point? @AliSoftware?

Reply:
Good idea on the docs then. It took me quite a while to understand the codebase a bit, and the documentation in the code is quite sparse.